### ISSUE TITLE
[Details Block] The space key doesn't work (Firefox)

### ISSUE THREAD

**Issue by gyurmey2** [WordPress/gutenberg#49679]
### Description

The space key doesn't work when entering text in the "Details Summary" field. This only happens in Firefox browsers.

### Step-by-step reproduction instructions

1. Add **Details** block in Firefox browser.
2. Go to the **Details Summary** field.
3. Try entering any text including spaces.

### Screenshots, screen recording, code snippet

![details](https://user-images.githubusercontent.com/113413673/230845286-6eb0b5ab-efd7-40dd-bc4c-0c9d5ea7ffa6.png)

### Environment info

- WordPress: 6.2
- Theme: TT3
- Browsers: **Firefox** 111.0.1, **Firefox Developer** 112.0b9, **Firefox Nightly** 113.0a1
- OS: Windows 10

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
@gyurmey2 
Tahnks for the rerport. I was able to reproduce it too.

Firefox Version: 111.0.1 (64-bit)

https://user-images.githubusercontent.com/54422211/230901974-056cd9ee-f53b-4779-befb-367a04409ef1.mp4

**Comment by t-hamano** [comment]
I think this is a problem unique to Firefox. If I give the `summary` element, or any of its children, the `contententeditable` attribute and press the space key in the text, it is treated as a click event and no space is inserted.

Here is the code I used to test:

```html
<!DOCTYPE html>
<html lang="ja">
<head>
<meta charset="UTF-8">
<title>Document</title>
</head>
<body>

<details>
<summary><div contenteditable>Editable Details Summary</div></summary>
<p>Details Content</p>
</details>

<script>
document.querySelector('summary').addEventListener('click', function(event) {
	console.log( 'clicked!' );
});
</script>

</body>
</html>
```

**Comment by carolinan** [comment]
I can reproduce this and I have tried to use different combinations of "KeyUp" instead of "onClick" for the _prevent_, but I have been unable to solve it.
I tried both with the separate `<summary>` element as a wrapper for the RichText (the way the block is currently built) and using summary as a tag in the RichText.

**Comment by talldan** [comment]
Here's the bug report for firefox - https://bugzilla.mozilla.org/show_bug.cgi?id=1822860. It has been triaged and accepted which is good.

I'm not sure if we can add any workarounds to RichText, or if we should just wait for the firefox bug to be resolved. It seems like it's being considered a normal priority, but I left a comment asking why that is and if there's a workaround.

**Comment by talldan** [comment]
There's a suggestion for a workaround from a Firefox maintainer:
> Currently, the workaround is not use contenteditable in the elements or prevent default of space keydown events and do document.execCommand("insertText", false, " ").

The latter sounds like a good option 👍

**Comment by t-hamano** [comment]
I am trying to fix in #50540.
